server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -299,35 +339,166 @@ private void updateToLoadAndDrop(List<Notice> notices, Map<String, LookupExtract
   private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap)
   {
     if (lookupSnapshotTaker != null) {
-      List<LookupBean> lookups = new ArrayList<>(lookupMap.size());
-      for (Map.Entry<String, LookupExtractorFactoryContainer> e : lookupMap.entrySet()) {
-        lookups.add(new LookupBean(e.getKey(), null, e.getValue()));
+      lookupSnapshotTaker.takeSnapshot(getLookupBeanList(lookupMap));
+    }
+  }
+
+  private void loadAllLookupsAndInitStateRef()
+  {
+    List<LookupBean> lookupBeanList = getLookupsListFromLookupConfig();
+    if (lookupBeanList != null) {
+      ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+      executorService = MoreExecutors.listeningDecorator(
```

Comment:
Please make it local variable, not a field.

Reply:
executorService is being reused in multiple methods. Why should we make it a local variable?

Reply:
I have made changes as suggested. Please review.